extensions/2.0/Vendor/EXT_texture_astc/README.md

```diff
@@ -0,0 +1,236 @@
+        },
+        {
+            "mimeType": "image/ktx2",
+            "bufferView": 2
+        }
+    ]
+}
+```
+
+### Using Without a Fallback
+
+To use KTX v2 image with ASTC compression without a fallback, define `EXT_texture_astc` in both `extensionsUsed` and `extensionsRequired`. The `texture` object will then have its `source` property omitted as shown below.
+
+```json
+{
+    "asset": {
+        "version": "2.0"
+    },
+    "extensionsUsed": [
+        "EXT_texture_astc"
+    ],
+    "extensionsRequired": [
+        "EXT_texture_astc"
+    ],
+    "textures": [
+        {
+            "extensions": {
+                "EXT_texture_astc": {
+                    "source": 0
+                }
+            }
+        }
+    ],
+    "images": [
+        {
+            "uri": "astc_image.ktx2"
+        }
+    ]
+}
+```
+
+### JSON Schema
+
+[texture.EXT_texture_astc.schema.json](schema/texture.EXT_texture_astc.schema.json)
+
+## KTX v2 files with ASTC compression 
+
+To cover a broad range of use cases, this extension allows different ASTC block sizes for LDR image types while HDR and 3D ASTC images are not supported. These can be determined from the vkFormat of the KTX image.
+
+### KTX header fields for ASTC payloads
+  - ASTC block sizes can be determined from the vkFormat of the KTX image. It will always be an LDR ASTC image.
+  - DFD `colorModel` MUST be `KHR_DF_MODEL_ASTC`.
+  - DFD `channelId` MUST be `KHR_DF_CHANNEL_ASTC_DATA`.
+
+### Additional requirements
+
+Regardless of the format used, these additional restrictions apply for compatibility reasons:
+
+- Orientation metadata (`KTXorientation`) MUST be `rd` or omitted.
+- Color space information in the DFD MUST match the expected usage, namely:
+  - For textures with **color data** (e.g., base color maps),
+    - `colorPrimaries` MUST be `KHR_DF_PRIMARIES_BT709`;
+    - `transferFunction` MUST be `KHR_DF_TRANSFER_SRGB`.
+  - For textures with **non-color data** (e.g., normal maps),
+    - `colorPrimaries` MUST be `KHR_DF_PRIMARIES_UNSPECIFIED`;
+    - `transferFunction` MUST be `KHR_DF_TRANSFER_LINEAR`.
+- When a texture refers to a sampler with mipmap minification, the KTX image MUST contain a full mip pyramid.
+- When a texture referencing a KTX v2 file with ASTC compressed image is used for glTF 2.0 material maps (both color and non-color), the KTX v2 image MUST be of **2D** type as defined in the KTX v2 Specification, Section 4.1.
+- `KHR_DF_FLAG_ALPHA_PREMULTIPLIED` flag MUST NOT be set unless the material's specification requires premultiplied alpha.
+
+## Known Implementations
+
+Authoring:
+
+TBA
+
+
+Viewing:
+
+TBA
+
+
+## Resources
+
+[KTX File Format Specification, version 2](https://github.khronos.org/KTX-Specification/)
+
+[KTX Reference Software](https://github.com/KhronosGroup/KTX-Software/)
+
+[ASTC encoder](https://github.com/ARM-software/astc-encoder)
+
+## Appendix: Full Khronos Copyright Statement
+
+Copyright 2023 The Khronos Group Inc.
```

Comment:
2025

Reply:
Fixed in https://github.com/KhronosGroup/glTF/pull/2369/commits/d1145f8f08440ae80ac407156d328d987c746557